### DISCUSSION TITLE
OpenWRT Package

### DISCUSSION THREAD

**Discussion by ranulphus** [bluenviron/mediamtx#319] [General]
## Which version are you using?

Latest

## Which operating system are you using?

OpenWRT

## Enhancement request

This would be fantastic in OpenWRT. Would it be difficult to create a package to cross-compile?

**Comment by aler9** [comment]
Hello, i did some research about doing a OpenWRT port.

Making OpenWRT packages of Golang-based programs is fully supported, and a couple ones are already in the main repository, like

* https://github.com/openwrt/packages/blob/master/net/nextdns/Makefile
* https://github.com/openwrt/packages/blob/master/net/dnstap/Makefile

`rtsp-simple-server` is compiled in the exact same way as `nextdns` and `dnstap`, so it should be enough to copy and edit the Makefile of these two and use it with the OpenWRT build system.

The package will probably work on the OpenWRT [targets](https://openwrt.org/docs/techref/targets/start) that the server already supports:
* x86_64
* x86
* arm64
* arm7
* arm6

The package will need testing on the targets that the servers doesn't support yet:
* powerpc
* mips
* arm < 6

When i'll have time to install the OpenWRT build system and QEmu, i'll write here the Makefile and instructions - of course external contributions are welcome.

[a comment by an automated account is not shown]
